casadm/cas_lib.c

```diff
@@ -547,6 +550,42 @@ void print_slow_atomic_cache_start_info(const char *device_path)
 	}
 }
 
+/**
+  * Save to dest an absolute device file path of src.
+  * Return number of characters copied to dest if succeed, negative value if failed.
+  */
+static int get_abs_path(char* dest, size_t dest_len, const char* src, size_t src_len)
+{
+	int path_len = -FAILURE;
+	char *dir_name, *dev_name;
+	char *dev = strndup(src, src_len);	// strdup creates hidden malloc
+	if (!dev)				// basename/dirname may modify the source and
+		goto dev_err;			// segfault when called with a static string
+
+	char *dir = strndup(src, src_len);
+	if (!dir)
+		goto dir_err;
+
+	dir_name = realpath(dirname(dir), NULL); // realpath creates hidden malloc
+	if (!dir_name)
+		goto dir_name_err;
+
+	dev_name = basename(dev);
+	if (!dev_name)
+		goto dev_name_err;
+
+	path_len = snprintf(dest, dest_len, "%s/%s", dir_name, dev_name);
+
```

Comment:
free dev_name here

Reply:
I checked it - dev_name isn't malloc'd, don't need to free. Compiler throws error.